### PR TITLE
 translate Chapter 7 (machine translation)

### PR DESCRIPTION
**Pull request by Haichao-Zhang**
Translation of Chapter 7 (machine translation) of the PaddlePaddle book.

### CODE REVIEW

File: machine_translation/README.en.md [hunk 1]
```diff
@@ -0,0 +1,697 @@
...
+
+The recent development of deep learning provides new solutions to those challenges. There are mainly two categories for deep learning based machine translation techniques: 1) techniques based on the statistical machine translation system but with some key components improved with neural networks, e.g., language model, reordering model (please refer to the left part of Figure 1); 2) techniques mapping from source language to target language directly using neural network, or end-to-end neural machine translation (NMT).
+
+<p align="center">
+<img src="image/nmt.png" width=400><br/>
+Figure 1. Neural Network based Machine Translation
```
Comment (wangkuiyi) [inline machine_translation/README.en.md]: Do we have Chinese text in this figure? If so, could you please list necessary translations here? This applies to all figures. Thanks!

File: machine_translation/README.en.md [hunk 2]
```diff
@@ -0,0 +1,697 @@
...
+</p>
+
+
+This tutorial will mainly introduce the NMT model and how to use PaddlePaddle to train an NMT model.
+
+## Illustrative Results
+
+Taking Chinese-to-English translation as an example, after training of the model, given the following segmented sentence in Chinese
+```text
+这些 是 希望 的 曙光 和 解脱 的 迹象 .
```
Comment (wangkuiyi) [inline machine_translation/README.en.md:28]: Should we also translate this Chinese sentence into English?

**Review by wangkuiyi — CHANGES_REQUESTED** [review]

File: machine_translation/README.en.md (hunk 2 above)
Comment (Haichao-Zhang) [reply machine_translation/README.en.md:28]: I guess not. As we are doing machine translation and the translated results  for this sentence are provided in the line immediately below it.

File: machine_translation/README.en.md (hunk 1 above)
Comment (Haichao-Zhang) [reply machine_translation/README.en.md]: Word translations added for all Figures. Note that there is an error in the original figure for Figure 4, as I already marked in its caption.

**Review by wangkuiyi — APPROVED** [review]

**Pull request merged**